From ae56a5dac36d5f39a57559738650cd97293a32e2 Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Wed, 20 Mar 2024 23:29:44 +0100
Subject: [PATCH 0872/1695] stm32mp2: add power domain device-tree bindings

Add power domain device-tree bindings for STM32MP25 SoC, and use it.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git ab6003e13a3571356980ba96ef8cb73bd8d9e795 stm32mp2: add power domain device-tree bindings]
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I220ce5fe5ad4f71c4de644f5d3164aa301f39370
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/373695
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/include/drivers/stm32mp_dt_bindings.h          |  1 +
 core/include/dt-bindings/power/st,stm32mp25-power.h | 11 +++++++++++
 2 files changed, 12 insertions(+)
 create mode 100644 core/include/dt-bindings/power/st,stm32mp25-power.h

diff --git a/core/include/drivers/stm32mp_dt_bindings.h b/core/include/drivers/stm32mp_dt_bindings.h
index 822af238d..fea7079ac 100644
--- a/core/include/drivers/stm32mp_dt_bindings.h
+++ b/core/include/drivers/stm32mp_dt_bindings.h
@@ -35,6 +35,7 @@
 #include <dt-bindings/soc/stm32mp25-rifsc.h>
 #include <dt-bindings/soc/stm32mp25-risab.h>
 #include <dt-bindings/soc/stm32mp25-risaf.h>
+#include <dt-bindings/power/st,stm32mp25-power.h>
 #include <dt-bindings/tamper/st,stm32mp25-tamp.h>
 #endif
 
diff --git a/core/include/dt-bindings/power/st,stm32mp25-power.h b/core/include/dt-bindings/power/st,stm32mp25-power.h
new file mode 100644
index 000000000..fe5d9f40c
--- /dev/null
+++ b/core/include/dt-bindings/power/st,stm32mp25-power.h
@@ -0,0 +1,11 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause) */
+/*
+ * Copyright (C) STMicroelectronics 2024 - All Rights Reserved
+ */
+
+#ifndef _DT_BINDINGS_POWER_STM32MP25_POWER_DOMAINS_H_
+#define _DT_BINDINGS_POWER_STM32MP25_POWER_DOMAINS_H_
+
+#define PD_SCMI_GPU             0
+
+#endif /* _DT_BINDINGS_POWER_STM32MP25_POWER_DOMAINS_H_ */
-- 
2.47.2

